### PR TITLE
feat: add `clickhouse` image

### PR DESCRIPTION
**Pull request by benfdking**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -90,9 +93,6 @@ pub mod trufflesuite_ganachecli;
 #[cfg(feature = "victoria_metrics")]
 #[cfg_attr(docsrs, doc(cfg(feature = "victoria_metrics")))]
 pub mod victoria_metrics;
-#[cfg(feature = "zookeeper")]
-#[cfg_attr(docsrs, doc(cfg(feature = "zookeeper")))]
-pub mod zookeeper;
```
Comment (DDtKey) [inline src/lib.rs:95]: Why zookeeper is removed? I guess not intentionally committed?

File: src/clickhouse/mod.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        // );
+        // let mut conn = postgres::Client::connect(connection_string, postgres::NoTls).unwrap();
+        //
+        // let rows = conn.query("SELECT 1 + 1", &[]).unwrap();
+        // assert_eq!(rows.len(), 1);
+        //
+        // let first_row = &rows[0];
+        // let first_column: i64 = first_row.get(0);
+        // assert_eq!(first_column, 2);
+        // Ok(())Ok
```
Comment (DDtKey) [inline src/clickhouse/mod.rs:135]: I suppose we can remove that

File: src/clickhouse/mod.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+impl Default for ClickHouseArgs {
+    fn default() -> Self {
+        Self {
+            command: "start-single-node".to_string(),
+            args: vec!["--insecure".to_string()],
+        }
+    }
+}
```
Comment (DDtKey) [inline src/clickhouse/mod.rs:66]: I think we should expose builder methods for these arguments.

Command can be hardcoded for now, but I think it's important to allow to override `insecure` at least

**Review by DDtKey — COMMENTED** [review]
Thank you for the contribution!

I've left couple of comments, please take a look

File: src/clickhouse/testdata/config.xml
Comment (DDtKey) [inline src/clickhouse/testdata/config.xml:1]: I see no usage of these files in testdata. Can we drop them?

**Comment by benfdking** [comment]
Hey, really sorry 🤦, I assumed this would create a file in our forked repo. It's not quite ready yet but trying to get it to it.

**Comment by DDtKey** [comment]
No worries, that's totally fine. We can convert PR to draft until it's ready for review 🙂

File: src/lib.rs (hunk 1 above)
Comment (benfdking) [reply src/lib.rs]: Readded.

File: src/clickhouse/mod.rs (hunk 3 above)
Comment (benfdking) [reply src/clickhouse/mod.rs]: This was copy pasta 🍝

File: src/clickhouse/testdata/config.xml
Comment (benfdking) [reply src/clickhouse/testdata/config.xml:1]: Removed ;)

**Comment by benfdking** [comment]
@DDtKey This should now be ready to go. Let me know if you have any comments/thoughts?

File: src/clickhouse/mod.rs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+
+    fn tag(&self) -> String {
+        self.tag.clone()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::seconds(10)]
```
[the remark this answers is not in the document]
Comment (benfdking) [reply src/clickhouse/mod.rs:62]:
```go
	WaitingFor: wait.ForAll(
			wait.NewHTTPStrategy("/").WithPort(httpPort).WithStatusCodeMatcher(func(status int) bool {
				return status == 200
			}),
		),
```

This is the strategy in Go. That would be better but yep we don't yet have that wait for method.

Comment (DDtKey) [inline src/clickhouse/mod.rs:62]:
```suggestion
        // TODO: use HTTP call to check rediness (depends on https://github.com/testcontainers/testcontainers-rs/issues/648)
        vec![WaitFor::seconds(10)]
```

File: src/clickhouse/mod.rs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        // query data
+        let query = "SELECT * FROM t";
+        let response = Client::new().post(url.clone()).body(query).send().await?;
+        assert_eq!(response.status(), 200);
+
+        // testing tcp endpoint
+        let client = clickhouse::Client::default().with_url(&format!("tcp://{}:{}", host, port));
+        #[derive(Row, Deserialize)]
+        struct MyRow {
+            a: u8,
```
Comment (DDtKey) [inline src/clickhouse/mod.rs:112]:
```suggestion
            #[serde(rename = "a")] // we don't read the field, so it's a dead-code in tests
            _a: u8,
```

**Review by DDtKey — APPROVED** [review]

**Comment by DDtKey** [comment]
Thank you!

**Comment by benfdking** [comment]
Thank you too! 🙏 Any chance you can do a new release? And do let me know when you build the http checker, happy to make the change!

**Comment by DDtKey** [comment]
Sorry for the delay, I completely forgot to release a new version.

I'm about to release `0.5.1` and HTTP verification will be available in test containers soon!

**Comment by benfdking** [comment]
Hey, please don't apologize! Thanks for stewarding the project and for the new update!

**Pull request merged**